paimon-core/src/main/java/org/apache/paimon/mergetree/compact/LookupChangelogMergeFunctionWrapper.java

```diff
@@ -136,12 +143,7 @@ public ChangelogResult getResult() {
         }
```

Comment:
actually `mergeFunction.candidates()` is the right order? maybe we can use it.

Reply:
The `mergeFunction.candidates()` are in the correct order, but we need to determine the right position when a new high-level record is found. Currently, it has not been compared using the user-defined sequence field or sequence number.

Reply:
See this [test](https://github.com/apache/paimon/pull/3286/files#diff-11928292ecbbd8b574d1a2329427e15330667b67d5bb979a151cf4f0843959f1R334). We have higher sequence in the high level value, the L0's seq is lower, if we add the high level value first, the result is wrong